### PR TITLE
[Experiment][TypeDeclaration] Allow change abstract class methods with body on ReturnType*Rector

### PR DESCRIPTION
**Pull request by samsonasik**
@TomasVotruba @staabm based on child return type sub type verification on previous PR:

- https://github.com/rectorphp/rector-src/pull/5343

This is experiment to allow abstract class methods with body to make return type as well on `ClassMethodReturnTypeOverrideGuard` service.

### CODE REVIEW

**Comment by samsonasik** [comment]
For change in vendor + non-vendor location, it already verified on `isReturnTypeChangeAllowed()` method 👍

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba @staabm it is ready for review.

**Comment by samsonasik** [comment]
Ok, It seems it will cause a bug on union:

```php
abstract class SomeAbstractWithUnion
{
    public function run()
    {
        if (rand(0, 1)) {
            return null;
        }

        return new \DateTime;
    }
}

final class NullableUnion extends SomeAbstractWithUnion
{
    public function run()
    {
        if (rand(0, 1)) {
            return null;
        }

        return new stdClass;
    }
}
```

it got:

```diff
 abstract class SomeAbstractWithUnion
 {
-    public function run()
+    public function run(): ?\DateTime
     {
         if (rand(0, 1)) {
             return null;
@@ @@
 
 final class NullableUnion extends SomeAbstractWithUnion
 {
-    public function run()
+    public function run(): ?\stdClass
     {
         if (rand(0, 1)) {
             return null;
```

on union type, it specifically on child verification tho, nothing todo with abstract, I will check in separate PR.

**Comment by samsonasik** [comment]
Oh, it seems fine, it because I wrongly copy paste class name, I will add the fixture

**Comment by samsonasik** [comment]
Added more fixtures :), Ready to review 👍

**Comment by samsonasik** [comment]
Ok, interface never have method with body, so I guess it can be rolled back for performance reason to skip it.

File: packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ public function shouldSkipClassMethod(ClassMethod $classMethod, Scope $scope): b
             return true;
         }
 
-        if ($classReflection->isAbstract()) {
+        if ($classMethod->isAbstract()) {
```
Comment (samsonasik) [inline packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php:48]: this is skip only on abstract method only, not abstract class 👍

**Comment by TomasVotruba** [comment]
@samsonasik Feel free to go for merge, let's try this in real world :wink:

**Comment by samsonasik** [comment]
Ok, let's merge it ;)

**Pull request merged**